### PR TITLE
[M6 Orders With Coupons] Expandable product cards in order creation - part 2

### PR DESCRIPTION
**Pull request by samiuelson**
## Expandable product cards in order creation - part 2
This is part 2 of the following PRs:
* Part 1: https://github.com/woocommerce/woocommerce-android/pull/9851 
* **Part 2: https://github.com/woocommerce/woocommerce-android/pull/9852**
* Part 3: https://github.com/woocommerce/woocommerce-android/pull/10018

### Description

This PR adds a basic expandable product cards UI, as per design: FTOWxedKUyDbBdOrxRfMyI-fi-14%3A9617.
The remaining, elements of the cards and various UI improvements are done in the next PR: https://github.com/woocommerce/woocommerce-android/pull/10018

### Testing instructions

Create a new order and add new products. Observe the new expandable product cards are rendered instead of the old ones.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4527432/fdfa194f-e3de-4f92-a9eb-5b45135f346d

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt [hunk 1]
```diff
@@ -475,33 +478,31 @@ class OrderCreateEditFormFragment :
...
+            productsSection.content = ComposeView(requireContext()).apply {
+                bindExpandableProductsSection(products)
             }
-            productsSection.content.productsAdapter?.apply {
-                submitList(products)
-                areProductsEditable = viewModel.currentDraft.isEditable
+        }
+    }
+
+    private fun ComposeView.bindExpandableProductsSection(items: LiveData<List<ProductUIModel>>) {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditFormFragment.kt:493]: Maybe to extra this to a separate file? The fragment already rather huge

And as a separate note, "OrderCreateEdit*" files probably should be in their own package

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    val transitionState = remember {
+        MutableTransitionState(isExpanded).apply { targetState = !isExpanded }
+    }
+    val transition = updateTransition(transitionState, "expandableProductCard")
+    val chevronRotation by transition.animateFloat(
+        transitionSpec = { tween(durationMillis = ANIM_DURATION_MILLIS) }, label = "chevronRotation"
+    ) {
+        if (isExpanded) 0f else 180f
+    }
+    Surface(modifier = Modifier.background(MaterialTheme.colors.surface)) {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: I think this is the wrong background color in the night mode. It used to match the card color

image

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+                    imageVector = Icons.Filled.KeyboardArrowDown,
+                    contentDescription =
+                    stringResource(R.string.order_creation_collapse_expand_product_card_content_description),
+                    tint = MaterialTheme.colors.primary
+                )
+            }
+            AnimatedVisibility(
+                visible = isExpanded,
+                modifier = Modifier
+                    .constrainAs(expandedPart) {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: I think the animation feels off. The elements go from the left top. Maybe to try make them appear from the top?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+                text = item.item.name,
+                modifier = Modifier
+                    .constrainAs(name) {
+                        top.linkTo(parent.top)
+                        start.linkTo(img.end)
+                    }
+                    .padding(dimensionResource(id = R.dimen.major_100))
+            )
+            IconButton(
+                onClick = { isExpanded = !isExpanded },
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: I'd consider expanding it not only when the icon is clicked, but on clicking on any place on the product

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+                bottom.linkTo(parent.bottom)
+                start.linkTo(parent.start)
+                end.linkTo(parent.end)
+            },
+            onClick = onRemoveProductClicked
+        ) {
+            Text(
+                text = stringResource(id = R.string.order_creation_remove_product),
+                color = colorResource(
+                    id = R.color.woo_red_70
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: woo_red_60 in the design:
```
background: #B32D2E;
```

image

Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt:281]: And the font doesn't look the same as on the design

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+                    }
+                    .size(dimensionResource(R.dimen.major_375))
+                    .padding(dimensionResource(id = R.dimen.major_100))
+                    .clip(RoundedCornerShape(dimensionResource(id = R.dimen.corner_radius_image)))
+            )
+            Text(
+                text = item.item.name,
+                modifier = Modifier
+                    .constrainAs(name) {
+                        top.linkTo(parent.top)
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: No right constraint?

image

**Review by kidinov — COMMENTED** [review]
I left a couple UI related suggestions, please take a look

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt (hunk 5 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: Renamed b740c95aa50aea6f9e03d73ecc8e8903f82f377a

Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: I actually thought that the color itself is not correct, but that was only the naming apparently. Thanks anyway

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt (hunk 6 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: Fixed here: 6850884d99de7ac95d8912541824826fe2ae83da
Also added max lines 1 and the end ellipsis:
Screenshot 2023-10-26 at 12 08 46

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt (hunk 3 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: Updated enter/exit transitions: dc8699081ce4112cec16070eee3b5c7657b20d31. Let me know what you think.
<video width= 300 src=https://github.com/woocommerce/woocommerce-android/assets/4527432/42d6a070-faa3-474b-b5fb-4e2c2f4d3f4d/>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt (hunk 2 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: Good catch. Fixed with 9261e3aa92e704c6f7c76745f1a7fe701b1e3654
I switched to using MaterialCardView's original background and made the compose background transparent, removing redundant Surface.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt (hunk 4 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/views/ExpandableProductCard.kt]: Done:
1. 691815300f3562c91b93b4325dba6cbc1529a422 set click listener
2. 8ba3ccfe8686d73aa288cc83caff45e3449ca10d removed ripple effect

**Comment by samiuelson** [comment]
> I left a couple UI related suggestions, please take a look

Thanks for the review. I've added the improvements. Let me know what you think!

**Review by kidinov — APPROVED** [review]
Thanks, @samiuelson . All changes are looking cool!

**Pull request merged**